src/TLuaInterpreter.cpp

```diff
@@ -12015,6 +12015,74 @@ int TLuaInterpreter::sendIrc( lua_State * L )
     return 0;
 }
 
+int TLuaInterpreter::setServerEncoding(lua_State * L)
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, "setServerEncoding: NULL Host pointer - something is wrong!");
+        return lua_error(L);
+    }
+
+    QString newEncoding;
+    if (! lua_isstring(L, 1)) {
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 type (newEncoding as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error( L );
+    }
+    else {
+        newEncoding = QString::fromUtf8(lua_tostring(L,1));
+    }
+
+    QPair<bool, QString> results = pHost->mTelnet.setEncoding(newEncoding);
+
+    if(results.first) {
+        lua_pushboolean(L, true);
```

Comment:
Bad change to Mudlets design philosophy - if I'd like to show a message to the screen along with my settings change, I would. Otherwise, this is a side-effect of me using the API that I wouldn't be able to get rid of since it's hardcoded behaviour :\

There are a lot of other ways people can shoot themselves in the foot and the API isn't in the business of getting into the script writers way all the time in an effort to save them.